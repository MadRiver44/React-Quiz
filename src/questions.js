// const questions = [
//   {
//     question: 'What is Redux?',
//     answers: [
//       'A state management tool',
//       'A tablet for indigestion',
//       'The newest version of JavaScipt',
//       'Nessecary for a React project',
//     ],
//     answerScore: '',
//   },
//   {
//     question: 'React was developed by:',
//     answers: [
//       'U.S government',
//       "React is a conspiracy, doesn't really exist",
//       'Facebook',
//       'Google',
//     ],
//   },
//   {
//     question: 'What is a pure function?',
//     answers: [
//       'A generic resusable function',
//       'A function that does not mutate its arguments nor produces side effects',
//       'A generator',
//       'Returns a long integer',
//     ],
//   },
//   {
//     question: 'A container component is:',
//     answers: [
//       'Handles the business logic and state and passes props to a presentational component',
//       'Same as a Presentational component',
//       'Invoked only once per react app',
//       'A node module',
//     ],
//   },
//   {
//     question: 'What does webpack do?',
//     answers: [
//       'Compiles sass and css files',
//       'Bundles all modules into a single file and performs optimizations',
//       'Packs an html and css file together',
//       'Creates a server for json files',
//     ],
//   },
// ];

const questions = [
  {
    question: 'What is Redux?',
    answers: [
      {
        answer: 'A state management tool',
        points: 3,
      },
      {
        answer: 'A tablet for indigestion',
        points: 0,
      },
      {
        answer: 'The newest version of JavaScipt',
        points: 2,
      },
      {
        answer: 'Nessecary for a React project',
        points: 1,
      },
    ],
    correctAnswer: 'answer1',
  },
  {
    question: 'React was developed by:',
    answers: [
      {
        answer: 'U.S government',
        points: 0,
      },
      {
        answer: "React is a conspiracy, doesn't really exist",
        points: 1,
      },
      {
        answer: 'Facebook',
        points: 3,
      },
      {
        answer: 'Google',
        points: 2,
      },
    ],
    correctAnswer: 'answer3',
  },
  {
    question: 'What is a pure function?',
    answers: [
      {
        answer: 'A generic resusable function',
        points: 2,
      },
      {
        answer: 'A function that does not mutate its arguments nor produces side effects',
        points: 3,
      },
      {
        answer: 'A generator',
        points: 1,
      },
      {
        answer: 'Returns a long integer',
        points: 0,
      },
    ],
    correctAnswer: 'answer2',
  },
  {
    question: 'A container component is:',
    answers: [
      {
        answer:
          'Handles the business logic and state and passes props to a presentational component',
        points: 3,
      },
      {
        answer: 'Same as a Presentational component',
        points: 2,
      },
      {
        answer: 'Invoked only once per react app',
        points: 1,
      },
      {
        answer: 'A node module',
        points: 0,
      },
    ],
    correctAnswer: 'answer1',
  },
  {
    question: 'What does webpack do?',
    answers: [
      {
        answer: 'Compiles sass and css files',
        points: 2,
      },
      {
        answer: 'Bundles all modules into a single file and performs optimizations',
        points: 3,
      },
      {
        answer: 'Packs an html and css file together',
        points: 1,
      },
      {
        answer: 'Creates a server for json files',
        points: 0,
      },
    ],
  },
];

export default questions;
