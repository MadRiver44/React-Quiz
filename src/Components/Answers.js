import React from 'react';
import questions from '../questions';

const Answers = props => {
  const { answer, handleClick } = props;
  let answersList = answer.map((a, index) => (
    <li key={index}>
      <button className="answers btn" onClick={e => handleClick(e)} value={a} id={a}>
        {a}
      </button>
    </li>
  ));
  return (
    <div>
      <ul>{answersList}</ul>
    </div>
  );
};

export default Answers;
