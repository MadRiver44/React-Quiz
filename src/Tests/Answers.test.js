import React from 'react';
import Answers from './../Components/Answers';
import questions from './../questions';
import renderer from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

describe('Answers component', () => {
  test('an array of answers renders correctly', () => {
    const tree = renderer
      .create(
        <Answers
          answer={[
            'A state management tool',
            'A tablet for indeigestion',
            'The newest version of JavaScipt',
            'Nessecary for a React project',
          ]}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('renders an empty list if none is provided', () => {
    const container = document.createElement('div');
    ReactDOM.render(<Answers answer={[]} />, container);
    expect(container.textContent).toMatch('');
  });
  test('answers array has only 4 choices', () => {
    const answersArray = [
      'A state management tool',
      'A tablet for indeigestion',
      'The newest version of JavaScipt',
      'Nessecary for a React project',
    ];
    expect(answersArray).toHaveLength(4);
  });
  test('correct answer is in answers array', () => {
    const answersArray = [
      'A state management tool',
      'A tablet for indeigestion',
      'The newest version of JavaScipt',
      'Nessecary for a React project',
    ];
    const correctAnswer = 'A state management tool';
    expect(answersArray).toContainEqual('A state management tool');
  });
});
