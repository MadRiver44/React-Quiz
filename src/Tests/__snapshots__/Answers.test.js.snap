// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answers component an array of answers renders correctly 1`] = `
<div>
  <form>
    <ul>
      <li>
        <button
          className="answers btn"
          onClick={[Function]}
          value="A state management tool"
        >
          A state management tool
        </button>
      </li>
      <li>
        <button
          className="answers btn"
          onClick={[Function]}
          value="A tablet for indeigestion"
        >
          A tablet for indeigestion
        </button>
      </li>
      <li>
        <button
          className="answers btn"
          onClick={[Function]}
          value="The newest version of JavaScipt"
        >
          The newest version of JavaScipt
        </button>
      </li>
      <li>
        <button
          className="answers btn"
          onClick={[Function]}
          value="Nessecary for a React project"
        >
          Nessecary for a React project
        </button>
      </li>
    </ul>
  </form>
</div>
`;
